cli/src/main/scala/com/codacy/analysis/cli/formatter/Formatter.scala

```diff
@@ -39,16 +39,19 @@ object Formatter {
   def apply(formatterName: String,
             executionDirectory: File,
             outputFile: Option[File] = Option.empty,
-            printStream: Option[PrintStream] = Option.empty): Formatter = {
-
-    val builder = allFormatters.find(_.name.equalsIgnoreCase(formatterName)).getOrElse {
-      logger.warn(s"Could not find formatter for name $formatterName. Using ${defaultFormatter.name} as fallback.")
-      defaultFormatter
-    }
+            printStream: Option[PrintStream] = Option.empty,
+            ghCodeScanningCompat: Boolean = false): Formatter = {
 
     val stream = outputFile.map(asPrintStream).orElse(printStream).getOrElse(defaultPrintStream)
 
-    builder(stream, executionDirectory)
+    formatterName match {
```

Comment:
I think if we start making special cases in the outside of the formatters it gets really bad.
What do you think about passing configurations to them and they decide if they use them?

Reply:
Yeah, you're right. I've changed things a bit to address this.